From 9bd292bd1fca41d438d6eaeedec04c23cc442d4f Mon Sep 17 00:00:00 2001
From: Raymond Toy <toy.raymond@gmail.com>
Date: Tue, 16 Apr 2019 19:17:32 -0700
Subject: [PATCH] Fix #68: Use -O1 when compiling with gcc now

As reported gcc 8.1.1 can't produce a working lisp.  gcc 8.3.1 also
fails.  But as reported on cmucl-imp, 2019-04-08, by Juan Pablo Hierro
Alverez, -O1 works.

Use -O1.
---
 src/lisp/Config.x86_common | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lisp/Config.x86_common b/src/lisp/Config.x86_common
index 31aebfc1e..d5537657b 100644
--- a/src/lisp/Config.x86_common
+++ b/src/lisp/Config.x86_common
@@ -44,7 +44,10 @@ endif
 
 CPPFLAGS := $(CPP_DEFINE_OPTIONS) $(CPP_INCLUDE_OPTIONS) 
 CFLAGS += -Wstrict-prototypes -Wall -g -fno-omit-frame-pointer
-CFLAGS += -O2
+
+# gcc 8.1.1 and 8.3.1 (and probably anything after 8.1.1?) won't
+# produce a working lisp with -O2.  Just use -O1.
+CFLAGS += -O1
 ASFLAGS = -g 
 
 ASSEM_SRC = x86-assem.S
-- 
2.21.0

